http-clients/src/main/java/org/triplea/http/client/web/socket/GenericWebSocketClient.java

```diff
@@ -53,7 +59,14 @@ public GenericWebSocketClient(
     client.addListener(this);
     CompletableFutureUtils.logExceptionWhenComplete(
         CompletableFuture.runAsync(client::connect, threadPool),
-        e -> log.log(Level.WARNING, "Failed to open connection with server", e));
+        e -> log.log(Level.SEVERE, "Failed to open connection with server", e));
+  }
+
+  @VisibleForTesting
+  static URI swapHttpsToWssProtocol(final URI uri) {
+    return uri.getScheme().equals("https")
+        ? URI.create(uri.toString().replace("https", "wss"))
+        : uri;
```

Comment:
Why aren't http links just converted to `ws://`?

Reply:
I'm not entirely sure. The http upgrade request is very happy over `http`, but is not for `https`. Since it's a secure connection, `wss` is needed, `ws` also does not work either (maybe we could write a forwarding rule for that, seems simpler for a number of reasons to go directly with the correct protocol here)

Reply:
The reason I went with an overwrite of the URI is to keep the 'override' configuration simpler, and also because this turned out to be a concern for only https.